api/objects.proto

```diff
@@ -222,3 +222,35 @@ message Cluster {
 	// a new key is allocated on key rotation.
 	uint64 encryption_key_lamport_clock = 6;
 }
+
+// Secret represents a secret that should be passed to a container or a node,
+// and is immutable. It wraps the `spec` provided by the user with useful
+// information that is generated from the secret data in the `spec`, such as
+// the digest and size of the secret data.
+message Secret {
```

Comment:
I prefer not to rely on the timestamps for behavior when possible, because the managers may have clocks that are slightly out of sync, so it would be possible to create a new secret version that is considered older than existing one.